quantization/nlp/bert/migraphx/e2e_migraphx_bert_example.py

```diff
@@ -0,0 +1,484 @@
+
+        self.enum_data_dicts = iter(data)
+        self.start_of_new_stride = True
+        return next(self.enum_data_dicts, None)
+
+def get_predictions(example_id_in_current_stride,
+                    features_in_current_stride,
+                    token_list_in_current_stride,
+                    batch_size,
+                    outputs,
+                    _NetworkOutput,
+                    all_predictions):
+                    
+    if example_id_in_current_stride == []:
+        return 
+
+    base_feature_idx = 0
+    for idx, id in enumerate(example_id_in_current_stride):
+        features = features_in_current_stride[idx]
+        doc_tokens = token_list_in_current_stride[idx]
+        networkOutputs = []
+        for i in range(len(features)):
+            x = (base_feature_idx + i) // batch_size
+            y = (base_feature_idx + i) % batch_size
+
+            output = outputs[x]
+            start_logits = output[0][y]
+            end_logits = output[1][y]
+
+            networkOutputs.append(_NetworkOutput(
+                start_logits = start_logits,
+                end_logits = end_logits,
+                feature_index = i 
+                ))
+
+        base_feature_idx += len(features) 
+
+        # Total number of n-best predictions to generate in the nbest_predictions.json output file
+        n_best_size = 20
+
+        # The maximum length of an answer that can be generated. This is needed
+        # because the start and end predictions are not conditioned on one another
+        max_answer_length = 30
+
+        prediction, nbest_json, scores_diff_json = dp.get_predictions(doc_tokens, features,
+                networkOutputs, n_best_size, max_answer_length)
+
+        all_predictions[id] = prediction
+
+def inference(data_reader, ort_session, latency, verbose=False):
+
+    _NetworkOutput = collections.namedtuple(  # pylint: disable=invalid-name
+            "NetworkOutput",
+            ["start_logits", "end_logits", "feature_index"])
+    all_predictions = collections.OrderedDict()
+    
+    example_id_in_current_stride = [] 
+    features_in_current_stride = []  
+    token_list_in_current_stride = []
+    outputs = []
+    while True:
+        inputs = data_reader.get_next(verbose)
+        if not inputs:
+            break
+
+        if data_reader.start_of_new_stride:
+            get_predictions(example_id_in_current_stride, features_in_current_stride, token_list_in_current_stride, data_reader.batch_size, outputs, _NetworkOutput, all_predictions)
+
+            # reset current example stride
+            example_id_in_current_stride = data_reader.example_id_list[-data_reader.example_stride:]
+            features_in_current_stride = data_reader.features_list[-data_reader.example_stride:] 
+            token_list_in_current_stride = data_reader.token_list[-data_reader.example_stride:]
+            outputs = []
+
+        io_binding = session.io_binding()
```

Comment:
this line works because python can deduce `session` variable, but this should use the input parameter name I guess:
`session` -> `ort_session`

Reply:
done